cid.go

```diff
@@ -133,6 +133,15 @@ var CodecToStr = map[uint64]string{
 // compatibility with the plain-multihash format used used in IPFS.
 // NewCidV1 should be used preferentially.
 func NewCidV0(mhash mh.Multihash) Cid {
+	// Need to make sure hash is valid for CidV0 otherwise we will
+	// incorrectly detect it as CidV1 in the Version() method
+	dec, err := mh.Decode(mhash)
+	if err != nil {
+		panic(err)
+	}
+	if dec.Code != mh.SHA2_256 || dec.Length != 32 {
+		panic("invalid hash for cidv0")
+	}
 	return Cid{string(mhash)}
```

Comment:
@Stebalien I panic here in order to avoid introducing another API change.  However, the patch to strict CIDs  (#40) does change the return type to `(Cid, error)` for both NewCidV0 and NewCidV1.  Should we just change them and get it now and get it over with?

I don't have strong felling either way?

Reply:
Okay, the only reason I suggested doing it now is to avoid another API change in the near future.  But we can keep this simple and change the return type later.

Reply:
I just don't want to drag this out.